argilla/src/argilla/_helpers/_deploy.py

```diff
@@ -0,0 +1,185 @@
+                        key=secret["key"],
+                        value=secret["value"],
+                        description=secret["description"],
+                    )
+
+                if space_hardware:
+                    hf_api.request_space_hardware(repo_id=repo_id, hardware=space_hardware)
+
+                if space_storage:
+                    hf_api.request_space_storage(repo_id=repo_id, storage=space_storage)
+                else:
+                    cls._space_storage_warning()
+        else:
+            if space_storage is None:
+                cls._space_storage_warning()
+
+            hf_api.duplicate_space(
+                from_id=_ARGILLA_SPACE_TEMPLATE_REPO,
+                to_id=repo_id,
+                private=private,
+                exist_ok=True,
+                hardware=space_hardware,
+                storage=space_storage,
+                secrets=secrets,
+            )
+
+        repo_url: RepoUrl = hf_api.create_repo(repo_id=repo_id, repo_type="space", exist_ok=True, space_sdk="docker")
+        api_url: str = (
+            f"https://{cls._sanitize_url_component(org_name)}-{cls._sanitize_url_component(repo_name)}.hf.space/"
+        )
+        cls._log_message(cls, message=f"Argilla is being deployed at: {repo_url}")
+        while cls._check_if_running(hf_api.get_space_runtime(repo_id=repo_id).stage):
+            time.sleep(_SLEEP_TIME)
+            cls._log_message(cls, message=f"Deployment in progress. Waiting {_SLEEP_TIME} seconds.")
+
+        headers = {}
+        if private:
+            headers["Authorization"] = f"Bearer {hf_token}"
+
+        return cls(api_url=api_url, api_key=api_key, headers=headers)
+
+    @staticmethod
+    def _space_storage_warning() -> None:
+        warnings.warn(
+            "No storage provided. The space will not have persistant storage so every 48 hours your data will be reset."
+        )
+
+    @classmethod
+    def _acquire_hf_token(cls, ht_token: Union[str, None]) -> str:
+        """Obtain the Hugging Face authentication token to deploy a space and authenticate."""
+        if ht_token is None:
+            ht_token = get_token()
+        if ht_token is None:
+            if cls._is_interactive():
+                notebook_login()
+            else:
+                login()
+            ht_token = get_token()
+        return ht_token
+
+    @classmethod
+    def _check_if_running(cls, stage: SpaceStage) -> bool:
+        """Check the current stage of the space runtime. Simplified to return True when being built."""
+        if stage in ["RUNNING"]:
+            return False
+        elif stage in ["RUNNING_APP_STARTING", "RUNNING_BUILDING", "BUILDING", "APP_STARTING"]:
+            return True
+        elif stage in ["PAUSED", "STOPPED"]:
+            return False
+        else:
+            raise ValueError(f"Space configuration is wrong and in stage: {stage}")
+
+    @classmethod
+    def _check_if_stage_can_be_build(cls, stage: SpaceStage) -> bool:
```

Comment:
This method only checks if the space is paused or stopped. So, maybe change the name to what it checks instead of what flow will do when it is stopped or paused.
```suggestion
    def _is_space_stopped(cls, stage: SpaceStage) -> bool:
```